contracts/MechMarketplace.sol

```diff
@@ -913,7 +933,7 @@ contract MechMarketplace is IErrorsMarketplace {
         uint256 mechServiceId = IMech(mech).tokenId();
 
         // Check mech validity as it must be created and recorded via this marketplace
-        if (mapServiceIdMech[mechServiceId] != mech) {
+        if (mapMechServiceIds[mech] != mechServiceId) {
```

Comment:
a malicious fake `mech` (that is not even registered in the system) can return `0` for `IMech(mech).tokenId()` (`mechServiceId`) and thus bypass the `checkMech(mech)` check.

Reply:
currently one can even simplify the current implementation (since request now provide the mech and not the service id) and get rid of `mapMechServiceIds` and instead just check for `mapAgentMechFactories[mech]` and make sure it is not `address(0)` and in the `remove(mech)` you clear this mapping instead.

Reply:
Good point, fixing